cmd/batch.go

```diff
@@ -360,6 +361,7 @@ func (b *Batch) WriteTemplate(
 	}
 
 	if err := b.TemplateCache.Execute(template, b.writer, templateData); err != nil {
+		os.Remove(b.versionFilePath)
```

Comment:
This only handles errors writing templates, but also now this function both writes templates and potentially deletes a file. Which will break in dry run as well as being confusing.

Reply:
The main reason I chose to put it there is this kind of error can happen any time a template is executed and I didn't want to pollute every error check for the `WriteTemplate` call in the batch command with an `os.Remove` call.  Since every error check that really matters for this was after a call to the `WriteTemplate` func it seemed like the place to put it.

> Which will break in dry run as well as being confusing.

I admit that I didn't even think about dry run here but I could just wrap the `os.Remove` call in a condition for dry run so it's more explicit.  Also, the next comment has more info on this...

> you are not handling the case of removing failing either, which could be why the tests still pass.

I can handle the error there and see if that makes a difference but I'm not sure it will.  Really the only errors I'd get from it are the file still being open(Windows) or it doesn't exist.  Which, the file not existing is what I ultimately want.  May have to close the file first before removing it for a Windows build.

What would you recommend be done here?

Reply:
@Magic-Mayo done with my suggestion, going to leave this open to let you take a peek but will merge it tomorrow after work if this satisfies your request.